plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableDataStore.php

```diff
@@ -802,85 +823,98 @@ private function generate_select_clause_for_props( $table_alias, $props ) {
 	 *
 	 * @param \WC_Order $order        The order.
 	 * @param boolean   $only_changes Whether to persist all order data or just changes in the object.
-	 * @return void
+	 * @return int      Order ID.
+	 * @throws \Exception If order data is not valid.
 	 */
 	protected function persist_order_to_db( $order, $only_changes = true ) {
```

Comment:
what do you think about refactoring this method a bit so that:
1. Instead of manually building insert/update data for each table inline, we generate it via another method. This way, when other plugins also add their table in the future, they can use a filter in that method to add additional rows that need to be persisted? 
2. Either way, this method currently does a lot, it would be good to split this in smaller chunks so that we can test it bit easily.

Reply:
Thanks for the suggestion @vedanshujain! I ended up simplifying this method quite a bit (despite adding support for inserting records into the posts table).
Rows for COT tables are now generated in a different function, which also allows for some extensibility using a filter.